src/Aspire.Dashboard/Protos/Partials.cs

```diff
@@ -51,11 +51,11 @@ ImmutableArray<EndpointViewModel> GetEndpoints()
                 .ToImmutableArray();
         }
 
-        static T ValidateNotNull<T>(T value, [CallerArgumentExpression(nameof(value))] string? expression = null) where T : class
+        T ValidateNotNull<T>(T value, [CallerArgumentExpression(nameof(value))] string? expression = null) where T : class
         {
             if (value is null)
             {
-                throw new InvalidOperationException($"Message field '{expression}' cannot be null.");
+                throw new InvalidOperationException($"Message field '{expression}' on resource with name '{Name}' cannot be null.");
```

Comment:
nit. Reads a bit better and I think it's clear what the name means here.
```suggestion
                throw new InvalidOperationException($"Message field '{expression}' on resource '{Name}' cannot be null.");
```

Reply:
The reason I wrote it like this is that the method also validates the `Name` property, so making it explicit is clearer if the name was null, to me at least.